src/util/base58.rs

```diff
@@ -83,14 +95,28 @@ pub fn from_check(data: &str) -> Result<Vec<u8>, Error> {
 }
 
 /// Encodes `data` as a base58 string.
+#[deprecated(since = "0.30.0", note = "Use base58::encode() instead")]
 pub fn encode_slice(data: &[u8]) -> String {
+    encode(data)
+}
+
+/// Encodes `data` as a base58 string (see also `base58::encode_check()`).
+pub fn encode(data: &[u8]) -> String {
     encode_iter(data.iter().cloned())
 }
 
 /// Encodes `data` as a base58 string including the checksum.
 ///
 /// The checksum is the first 4 256-digits of the object's Bitcoin hash, concatenated onto the end.
+#[deprecated(since = "0.30.0", note = "Use base58::encode_check() instead")]
 pub fn check_encode_slice(data: &[u8]) -> String {
+    encode_check(data)
+}
+
+/// Encodes `data` as a base58 string including the checksum.
+///
+/// The checksum is the first 4 256-digits of the object's Bitcoin hash, concatenated onto the end.
```

Comment:
In e201e201c84495016864ecef6d9365f0a6ca7c1d:

Heh, this comment is pretty badly written. We should change it to "the first four bytes of the sha256d of the data". It's not a "bitcoin hash" and it's not an "object" that we're hashing.

Reply:
Ha, I should have more confidence in my Bitcoin knowledge, I thought this was odd and figured I just didn't know enough. Will fix.